### PR TITLE
net: golioth: system_client: Added ability to manage cloud interaction.

### PR DESCRIPTION
**Pull request by ubieda**
- Added mechanism for stopping system-client upon
golioth_system_client_stop() call.
- Added flag FLAG_STOP_CLIENT to signal the system-client that the user
wants to suspend the interactions with the cloud backend.
- Added hello_sporadic sample that portrays the configuration and API
usage to implement this functionality.

This is a follow-up PR of #29, as due to Golioth's Zephyr SDK repo name
changed, my forked repo lost track of updates, even after rebasing them
locally. Having said that, this PR addresses #25, so I suggest we close that
one and continue the discussion on this one.

@mniestroj, feel free to check it out and leave your comments!

<a href="https://gitpod.io/#https://github.com/golioth/zephyr-sdk/pull/140"></a>

### CODE REVIEW

**Comment by beriberikix** [comment]
Thanks @ubieda! In parallel to reviewing the code, let's fix one of the checks around [DCO](https://github.com/golioth/zephyr-sdk/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-process). We recently added this check, following Zephyr's model. Can you sign off the contribution?

**Comment by ubieda** [comment]
@beriberikix Gotcha, no problem. I believe the commit message is already signed-off. Should it be signed-off differently or put elsewhere?

**Comment by ubieda** [comment]
Oh - I realize what happened: I signed off with my croxel email. I'll change it to match the one from github account.

File: net/golioth/system_client.c [hunk 1]
```diff
@@ -283,14 +299,17 @@ static void golioth_system_client_main(void *arg1, void *arg2, void *arg3)
...
 			/*
-			 * Reconnect request is handled similar to recv timeout.
+			 * Reconnect and stop requests are handled similar to recv timeout.
 			 */
-			if (reconnect_request || receive_timeout) {
+			if (reconnect_request || receive_timeout || stop_request) {
 				if (reconnect_request) {
 					LOG_INF("Reconnect per request");
+				} else if (stop_request) {
+					LOG_INF("Stop request");
```
Comment (mniestroj) [inline net/golioth/system_client.c:312]: Minor nit. Assuming that both "stop" and "reconnect" will be handled at the same time, then the "stop" will take precedence. So I would reorder if statements here and do:
```suggestion
				if (stop_request) {
					LOG_INF("Stop request");
				} else if (reconnect_request) {
					LOG_INF("Reconnect per request");
```

**Review by mniestroj — COMMENTED** [review]
Looks good overall!

File: net/golioth/system_client.c (hunk 1 above)
Comment (ubieda) [reply net/golioth/system_client.c]: Good point. Change addressed!

File: samples/hello_sporadic/README.rst [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+nRF9160 Feather
+---------------
+
+On your host computer open a terminal window, locate the source code of this
+sample application (i.e., ``samples/hello_sporadic``) and type:
+
+.. code-block:: console
+
+   $ west build -b circuitdojo_feather_nrf9160ns samples/hello_sporadic
```
Comment (mniestroj) [inline samples/hello_sporadic/README.rst:139]: This should be now `circuitdojo_feather_nrf9160ns`. Notice `_ns` on the end. The same goes for overlay file name.

**Review by mniestroj — CHANGES_REQUESTED** [review]
One last request and we are good to merge.

File: samples/hello_sporadic/README.rst (hunk 2 above)
Comment (ubieda) [reply samples/hello_sporadic/README.rst]: Understood. Both were changed to `circuitdojo_feather_nrf9160_ns` now

**Review by mniestroj — APPROVED** [review]

**Pull request merged**